packages/react-native-editor/__device-tests__/helpers/utils.js

```diff
@@ -452,6 +449,29 @@ const toggleOrientation = async ( driver ) => {
 	}
 };
 
+const isEditorVisible = async ( driver, iteration = 0 )  => {
+	if ( iteration >= 25 ) {
+		throw new Error(
+			`Gutenberg Editor is still not visible after ${ iteration } retries!`
+		);
+	} else if ( iteration !== 0 ) {
+		// wait 1 second before trying to locate element again
+		await driver.sleep( 1000 );
+	}
+
+	const blockLocator = isAndroid()
+		? `//android.widget.EditText[@content-desc="Post title. Welcome to Gutenberg!"]`
+		: `(//XCUIElementTypeScrollView/XCUIElementTypeOther/XCUIElementTypeOther[@name="Post title. Welcome to Gutenberg!"])`;
+
+	const locator = await driver.elementsByXPath( blockLocator );
+
+	if ( locator.length !== 1 ) {
+		// if locator is not visible, try again
+		iteration++;
+		return await isEditorVisible( driver, iteration );
```

Comment:
Nitpick: I think we could simplify this by doing:
```suggestion
		return await isEditorVisible( driver, iteration + 1 );
```